syntax = "proto3";

service BookService {
    rpc GetBooks (Empty) returns (BookList);
    rpc GetBookById (BookRequest) returns (Book);
}

message Empty {}

message Book {
    int32 id = 1;
    string title = 2;
    string author = 3;
}

message BookRequest {
    int32 id = 1;
}

message BookList {
    repeated Book books = 1;
}
